File: pkg/pb/time.go
package pb

import (
	"fmt"

	"google.golang.org/protobuf/types/known/timestamppb"
)

// Format the sclk to a string
func (t *Time) SclkS() string {
	return fmt.Sprintf("%.06f", t.Sclk)
}

// Check if this timestamp has a sclk attached to it
func (t *Time) HasSclk() bool {
	if t == nil {
		return false
	}

	return t.GetSclk() > 0
}

func TimeNow() *Time {
	return &Time{
		Unix: timestamppb.Now(),
	}
}

File: src/modules/rpc/src/proto/hermes/hermes/Time.ts
// Original file: proto/pb/time.proto

import type { Timestamp as _google_protobuf_Timestamp, Timestamp__Output as _google_protobuf_Timestamp__Output } from '../google/protobuf/Timestamp';

export interface Time {
  'unix'?: (_google_protobuf_Timestamp | null);
  'sclk'?: (number | string);
}

export interface Time__Output {
  'unix'?: (_google_protobuf_Timestamp__Output);
  'sclk'?: (number);
}
